node-src/tasks/prepare.ts

```diff
@@ -0,0 +1,206 @@
+import * as turbosnap from '@cli/turbosnap';
+import { readdirSync, readFileSync, statSync } from 'fs';
+import path from 'path';
+import semver from 'semver';
+import slash from 'slash';
+
+import { getFileHashes } from '../lib/getFileHashes';
+import { createTask, transitionTo } from '../lib/tasks';
+import { rewriteErrorMessage } from '../lib/utils';
+import { Context, Task } from '../types';
+import missingStatsFile from '../ui/messages/errors/missingStatsFile';
+import deviatingOutputDirectory from '../ui/messages/warnings/deviatingOutputDirectory';
+import {
+  bailed,
+  hashing,
+  initial,
+  invalid,
+  success,
+  traced,
+  tracing,
+  validating,
+} from '../ui/tasks/prepare';
+
+interface PathSpec {
+  pathname: string;
+  contentLength: number;
+}
+
+// These are the special characters that need to be escaped in the filename
+// because they are used as special characters in picomatch
+const SPECIAL_CHARS_REGEXP = /([$()*+?[\]^])/g;
+
+// Get all paths in rootDir, starting at dirname.
+// We don't want the paths to include rootDir -- so if rootDir = storybook-static,
+// paths will be like iframe.html rather than storybook-static/iframe.html
```

Comment:
Since this is an API (of sorts), thoughts on making this JSDoc comment so that we can get easy descriptions on mouseover in VSCode and other tools?